### DISCUSSION TITLE
Is it possible to auto generate CRUD endpoints by introspecting the database schema?

### DISCUSSION THREAD

**Discussion by kawpii** [PostgREST/postgrest#2797] [Q&A] [answered]
Automatically generating endpoints for create, read, update, delete for each table by introspecting the schema. User should be able to disable each CRUD operation by configuration file. This feature is available in Postgraphile by default.

Is this already available in PostgREST? I search the docs found nothing related to this.

**Comment by steve-chavez** [comment, marked as answer]
> Automatically generating endpoints for create, read, update, delete for each table by introspecting the schema. 

You mean all of this? https://postgrest.org/en/stable/references/api/tables_views.html

- read https://postgrest.org/en/stable/references/api/tables_views.html#read
- update https://postgrest.org/en/stable/references/api/tables_views.html#update
- ...

**Reply by kawpii** [reply]
Yes it is.
